dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3763,240 +3838,300 @@ Sedp::Reader::data_received(const DCPS::ReceivedDataSample& sample)
     }
     ser.encoding(encoding);
 
-    if (entity_id == ENTITYID_SEDP_BUILTIN_PUBLICATIONS_WRITER) {
-      ParameterList data;
-      if (!decode_parameter_list(sample, ser, extensibility, data)) {
-        ACE_ERROR((LM_ERROR, ACE_TEXT("ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to deserialize data\n")));
-        return;
-      }
+    data_received_i(sample, entity_id, ser, extensibility);
+    break;
+  }
 
-      DCPS::unique_ptr<DiscoveredPublication> wdata(new DiscoveredPublication);
-      if (!ParameterListConverter::from_param_list(data, wdata->writer_data_, wdata->type_info_)) {
-        ACE_ERROR((LM_ERROR,
-                   ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to convert from ParameterList ")
-                   ACE_TEXT("to DiscoveredWriterData\n")));
-        return;
-      }
-#ifdef OPENDDS_SECURITY
-      wdata->have_ice_agent_info_ = false;
-      ICE::AgentInfoMap ai_map;
-      if (!ParameterListConverter::from_param_list(data, ai_map)) {
-        ACE_ERROR((LM_ERROR,
-                   ACE_TEXT("(%P|%t) ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to convert from ParameterList ")
-                   ACE_TEXT("to ICE Agent info\n")));
-        return;
-      }
-      ICE::AgentInfoMap::const_iterator pos = ai_map.find("DATA");
-      if (pos != ai_map.end()) {
-        wdata->have_ice_agent_info_ = true;
-        wdata->ice_agent_info_ = pos->second;
-      }
-#endif
-      sedp_.task_.enqueue(id, move(wdata));
+  default:
+    break;
+  }
+}
 
-#ifdef OPENDDS_SECURITY
-    } else if (entity_id == ENTITYID_SEDP_BUILTIN_PUBLICATIONS_SECURE_WRITER) {
-      ParameterList data;
-      if (!decode_parameter_list(sample, ser, extensibility, data)) {
-        ACE_ERROR((LM_ERROR, ACE_TEXT("ERROR: Sedp::Reader::data_received - ")
-                   ACE_TEXT("failed to deserialize data\n")));
-        return;
-      }
+void
+Sedp::LivelinessReader::data_received_i(const DCPS::ReceivedDataSample& sample,
+  const DCPS::EntityId_t& entity_id,
+  DCPS::Serializer& ser,
+  const DCPS::Extensibility extensibility)
+{
+  ACE_UNUSED_ARG(extensibility);
```

Comment:
```suggestion
```
In this case simply remove the name from the method's parameter list.

Reply:
I am trying to keep same `virtual data_received_i()` function for all derived classes, as they are  called from the base class function.

Reply:
This is not related to parameter names.  The idea is that if you have function definition `void foo(int a) {}`, the language doesn't need the `a` to be there, since it's not used.  This avoids a warning and doesn't need an ACE MACRO.  The presence or absence of a name is not part of a function's type signature and doesn't impact callers or virtual overrides.